### PR TITLE
added experiments.py 

### PR DESCRIPTION
**Pull request by josh-marsh**
This change is [Reviewable](https://reviewable.io/reviews/matthewja/thursday/34)

### CODE REVIEW

**Comment by MatthewJA** [comment]
Next time, please make sure you're working off a new branch so we don't have all the old commits that have already been PR'd.

---

Review status: 0 of 7 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[thursday/experiments.py, line 2 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwT8KB2NT6zIoPKcH:-LAUwT8KB2NT6zIoPKcI:b-ds3rty) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L2)):*
> ```Python
> import h5py
> ```

Drop these newlines — h5py, numpy, and sklearn are all nonstandard modules.

---

*[thursday/experiments.py, line 8 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwW9d3u_t1lWCMgZQ:-LAUwW9d3u_t1lWCMgZR:bb8xnvp) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L8)):*
> ```Python
> from sklearn.linear_model import LogisticRegressionCV
> from sklearn.naive_bayes import GaussianNB
> ```

Just one newline here

---

*[thursday/experiments.py, line 13 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwYMKANkNLJ_vS4U3:-LAUwYMKANkNLJ_vS4U4:b3k5hv9) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L13)):*
> ```Python
> from models import SklearnModel, HOGNet
> 
> print ("Experiment 1: Classifing FR-I vs FR-II")
> ```

Don't put a space between print and (

---

*[thursday/experiments.py, line 34 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwfiy78cRbTqZvEcq:-LAUwfiy78cRbTqZvEcr:byp570c) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L34)):*
> ```Python
>     train_x = np.expand_dims(train_x, axis=3)
> 
> with h5py.File("data/data.h5", 'r') as data:
> ```

Why do you open the file twice? Delete this line.

---

*[thursday/experiments.py, line 44 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwkfJ79e6_ixPlYOG:-LAUwkfJ79e6_ixPlYOH:b7xhegw) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L44)):*
> ```Python
> # Instantiating Classifiers
> random_forest = SklearnModel(RandomForestClassifier, datagen=datagen, nb_augment=10, seed=seed)
> #logistic_reg = SklearnModel(LogisticRegressionCV, datagen=datagen, nb_augment=5, seed=seed)
> ```

If this isn't working, delete it.

---

*[thursday/experiments.py, line 47 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwm0r94a74hUlVDCw:-LAUwm0r94a74hUlVDCx:bcjdymj) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L47)):*
> ```Python
> naive_bayes = SklearnModel(GaussianNB, datagen=datagen, nb_augment=5, seed=seed)
> hognet = HOGNet(datagen=datagen, batch_size=64, steps_per_epoch=50, 
>                                 max_epoch=100, patience=5, gap=2, seed=seed)
> ```

`max_epoch` should line up with `datagen=datagen` (GS)

---

*[thursday/experiments.py, line 58 at r1](https://beta.reviewable.io/reviews/matthewja/thursday/34#-LAUwpVg2bXN412hTl_L:-LAUwpVg2bXN412hTl_M:b-l4aqan) ([raw file](https://github.com/matthewja/thursday/blob/2198d89d46cc88750e176bea2c6321412203bebc/thursday/experiments.py#L58)):*
> ```Python
> 
>     correct = test_y == predictions
>     print('Accuracy: {:.02%}'.format(correct.mean()))
> ```

Is this dataset class-balanced? I don't think it is if you're using the Aniyan test set, so make sure to also report the balanced accuracy here.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/matthewja/thursday/34#-:-LAUwIQF5iQmVAyn_7-8:b-kr8w7t)*

**Comment by MatthewJA** [comment]
Looks good with some nits. Fix those and I'll merge.

---

Review status: 0 of 7 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/matthewja/thursday/34#-:-LAUwzn6DTdoDwFhI7JD:b-er2r3z)*

**Comment by josh-marsh** [comment]
Made changes. And yes, Aniyan is class balanced.

**Comment by MatthewJA** [comment]
Consider making it not class-balanced so we can use the full training set available.

**Comment by MatthewJA** [comment]
Can you fix the conflicts and recommit?
